### PR TITLE
Tasks should be transmitted with DELETE requests, too

### PR DESCRIPTION
**Pull request by dsheets**
Without this, delete requests over HTTP use the previous store task
which is often the server store creation task. Requires mirage/ocaml-cohttp#383.

Some of the code in `Irmin_http_server` common to `POST` and `DELETE` processing could be factored.

### CODE REVIEW

**Comment by samoht** [comment]
a workaround is to not use DELETE and use POST everywhere ...

**Comment by dsheets** [comment]
It is true that `POST` could be used everywhere but the protocol semantics are slightly different. In particular, `DELETE` is idempotent whereas `POST` is not. I'm not sure it *really* matters but using `DELETE` for deletion makes sense and would be nice.

What would you like to do?

**Comment by samoht** [comment]
Depends if we can have a quick point release for cohttp or not. But I'm fine to use pinning.

File: lib/http/irmin_http.ml [hunk 1]
```diff
@@ -186,6 +181,12 @@ module Helper (Client: Cohttp_lwt.Client) = struct
     in
     short_body, body
 
+  let delete t ~task path fn =
+    let uri = uri_append t path in
+    let short_body, body = make_body ?task None in
+    Log.debug "delete %s %s" (Uri.path uri) short_body;
+    Client.delete ?body ~headers uri >>= map_string_response fn
```
Comment (samoht) [inline lib/http/irmin_http.ml]: Can you use `Client.call .... `DELETE` instead?

**Comment by samoht** [comment]
Thanks!

**Pull request merged**